### DISCUSSION TITLE
504 Error | Pro Member | Serverless

### DISCUSSION THREAD

**Discussion by space1worm** [vercel/next.js#56410] [Help]
### Summary

Hey everyone, 

**Next.js version : 13.5.3** 
**Dir: App Directory**

Do you have any ideas about why Vercel is returning a 504 error? I'm on the pro plan, and my serverless function is configured to run for a maximum of 5 minutes. Typically, my function for testing purposes needs 1.5m to execute

In the logs, **I can see that the serverless function is returning a 200 status code**, and it's executing correctly. However, when I access it through postman, after exactly 1 minute, it returns a **504 error.**

Everything works fine locally. What could be the issue?
<img width="1201" alt="Screenshot 2023-10-04 at 14 47 25" src="https://github.com/vercel/next.js/assets/84189246/bab7b72c-137d-44bd-9517-ed56ce5176e6">
<img width="1203" alt="Screenshot 2023-10-04 at 14 47 34" src="https://github.com/vercel/next.js/assets/84189246/768ceedd-efa8-446f-bf95-0825b0c40574">
<img width="837" alt="Screenshot 2023-10-04 at 14 50 28" src="https://github.com/vercel/next.js/assets/84189246/f9138858-0fef-4aac-be02-5436216255e3">
<img width="702" alt="Screenshot 2023-10-04 at 14 53 19" src="https://github.com/vercel/next.js/assets/84189246/37acdf8a-ade2-4faf-868c-518c85a58f3c">
<img width="708" alt="Screenshot 2023-10-04 at 14 53 27" src="https://github.com/vercel/next.js/assets/84189246/f1b21818-380a-405e-8533-3338fc327017">

**Comment by Patrikur** [comment]
Sorry to see that you did not get a response yet.
We encountered the exact same issue on version 14.0.1, and it came out of nowhere. The one day everything was working fine, the other the error occurs, but we did not touch it at all. Our logs are also not pointing us in any direction.
We changed to pages dir on a smaller project, and the issue was gone for that. But this cannot be the solution to it.
I asked as well in the Discord and will flag this as a bug to the dev team if this is not getting any useful solution.
